lib/util/cache/package/decorator.ts

```diff
@@ -66,21 +70,59 @@ export function cache<T>({
       return callback();
     }
 
-    const cachedResult = await packageCache.get<unknown>(
+    finalKey = `cache-decorator:${finalKey}`;
+    const oldRecord = await packageCache.get<DecoratorCachedRecord>(
       finalNamespace,
       finalKey
     );
 
-    if (cachedResult !== undefined) {
-      return cachedResult;
+    const softTtl = ttlMinutes;
+
+    const cacheHardTtlMinutes = GlobalConfig.get().cacheHardTtlMinutes ?? 0;
+    let hardTtl = softTtl;
+    if (methodName === 'getReleases' || methodName === 'getDigest') {
+      hardTtl = Math.max(softTtl, cacheHardTtlMinutes);
     }
 
-    const result = await callback();
+    let oldData: unknown;
+    if (oldRecord) {
+      const now = DateTime.local();
+      const cachedAt = DateTime.fromISO(oldRecord.cachedAt);
```

Comment:
wasn't one idea to store the cachedAt value as separate key, so we don't need to fetch the probably huge value if it's expired?

Reply:
It was another PR, where I wanted to store `updatedAt` in another key, though we completely deleted it as it wasn't actually used anywhere.

Reply:
would it make sense to do here?

Reply:
No, it will break everything